pkg/reconciler/pipelinerun/resources/apply.go

```diff
@@ -237,10 +237,11 @@ func ApplyTaskResultsToPipelineResults(
 	results []v1beta1.PipelineResult,
 	taskRunResults map[string][]v1beta1.TaskRunResult,
 	customTaskResults map[string][]v1alpha1.RunResult) []v1beta1.PipelineRunResult {
-
+	// TODO(#4723): Validate array and object results.
```

Comment:
Is this TODO for validating the substituted result matches the substituting result? I thought that would be in pipelinerun reconciler?